v-next/hardhat/src/internal/builtin-plugins/network-manager/hook-handlers/config.ts

```diff
@@ -104,7 +104,7 @@ export async function resolveUserConfig(
   return {
     ...resolvedConfig,
     chainDescriptors: await resolveChainDescriptors(
-      resolvedConfig.chainDescriptors,
+      userConfig.chainDescriptors,
```

Comment:
I’m not sure this is correct, but we’re doing the same thing with the networks. Shouldn’t we always use the resolved version to account for other hook handler changes? Doing so causes type issues, though, since I’d be resolving a field that’s already resolved.